### DISCUSSION TITLE
How to run math operations on Zarr

### DISCUSSION THREAD

**Discussion by meteoDaniel** [zarr-developers/zarr-python#1820] [Q&A] [answered]
Dear all, 

 I have created a native Zarr using `zarr.open()` and wrote data into it. 

Afterwards I want to run calculations on that zarr, in the same way as I would do with numpy but receive: 
``` python
zarr TypeError: unsupported operand type(s) for -: 'Array' and 'float'
``` 

Edit:
I want to avoid loading all the data into ram. 

Any thoughts how to work around that issue?

**Comment by dcherian** [comment, marked as answer]
You have to read the data first so eitehr `np.array(zarr_array) - 1` or `zarr_array[:] - 1`  (I think)

**Reply by meteoDaniel** [reply]
But in that case, all the data will be load into memory and that is not what I want.

**Reply by dcherian** [reply]
Then use `dask`. `dask.array.from_array(zarr_array) - 1`
